fsspec/implementations/arrow.py

```diff
@@ -139,12 +146,18 @@ def rm(self, path, recursive=False, maxdepth=None):
     @wrap_exceptions
     def _open(self, path, mode="rb", block_size=None, **kwargs):
         if mode == "rb":
-            stream = self.fs.open_input_stream(path)
+            method = self.fs.open_input_stream
         elif mode == "wb":
-            stream = self.fs.open_output_stream(path)
+            method = self.fs.open_output_stream
         else:
             raise ValueError(f"unsupported mode for Arrow filesystem: {mode!r}")
 
+        # Check if pyarrow detects compression
+        _kwargs = {}
+        if self._pyarrow_version >= version.parse('4.0'):
+            _kwargs["compression"] = None
```

Comment:
So 4.0+ will disable compression, but older ones will `detect` it, right? That seems like an odd behavioural change for fs to have.

Also, you mention `with fss.open('compressed.gz', compression='infer')` in the original issue, which raises the question of who should receive that arg. The ArrowFile(based on io.IOBase) doesn't seem to care for it, right? Should we catch this argument in fs.open then and not pass it through?

Reply:
It's the opposite way - starting at v4.0 pyarrow has a `compression='detect'` arg - which defaults to decompress.
if `fss.open('some.gz', compression='infer')` is used, it receives a stream which is already decompressed
if `fss.open('some.gz', compression=None)` is used, it receives a decompressed stream which is unintentional

I've added a code example in the ticket

Reply:
by `fss.open` you mean open in fsspec fs, right? If so, that `compression=` arg is not being passed to the method at all, hence why I'm wondering about the desired behaviour here and wether or not `fs.open` here should catch it.

Reply:
Ah, ok, there is misunderstanding here on my side. I see that it is being used down the line.